Compile error with pulse_counter

### The problem

When i try to build with 2026.2.0 the pulse_counter component isn't compiling any more. worked with 2026.1.x before.

I have a esp32-c6

### Which version of ESPHome has the issue?

2026.2.0

### What type of installation are you using?

Home Assistant Add-on

### What platform are you using?

ESP32

### Component causing the issue

pulse_counter

### YAML Config

```yaml
sensor:
  - platform: pulse_counter
    pin: GPIXX
    name: "Pulse Counter"
```

### Anything in the logs that might be useful for us?

```txt
src/esphome/components/pulse_counter/pulse_counter_sensor.cpp: In member function 'virtual bool esphome::pulse_counter::HwPulseCounterStorage::pulse_counter_setup(esphome::InternalGPIOPin*)':
src/esphome/components/pulse_counter/pulse_counter_sensor.cpp:121:34: error: 'SOC_MOD_CLK_APB' was not declared in this scope; did you mean 'SOC_MOD_CLK_CPU'?
  121 |     esp_clk_tree_src_get_freq_hz(SOC_MOD_CLK_APB, ESP_CLK_TREE_SRC_FREQ_PRECISION_CACHED, &apb_freq);
      |                                  ^~~~~~~~~~~~~~~
      |                                  SOC_MOD_CLK_CPU
*** [.pioenvs/xxx/src/esphome/components/pulse_counter/pulse_counter_sensor.cpp.o] Error 1
```

### Additional information

_No response_

--- a/esphome/components/pulse_counter/pulse_counter_sensor.cpp
+++ b/esphome/components/pulse_counter/pulse_counter_sensor.cpp
@@ -2,7 +2,7 @@
 #include "esphome/core/log.h"
 
 #ifdef HAS_PCNT
-#include <esp_clk_tree.h>
+#include <esp_private/esp_clk.h>
 #include <hal/pcnt_ll.h>
 #endif
 
@@ -117,9 +117,7 @@ bool HwPulseCounterStorage::pulse_counter_setup(InternalGPIOPin *pin) {
   }
 
   if (this->filter_us != 0) {
-    uint32_t apb_freq;
-    esp_clk_tree_src_get_freq_hz(SOC_MOD_CLK_APB, ESP_CLK_TREE_SRC_FREQ_PRECISION_CACHED, &apb_freq);
-    uint32_t max_glitch_ns = PCNT_LL_MAX_GLITCH_WIDTH * 1000000u / apb_freq;
+    uint32_t max_glitch_ns = PCNT_LL_MAX_GLITCH_WIDTH * 1000000u / (uint32_t) esp_clk_apb_freq();
     pcnt_glitch_filter_config_t filter_config = {
         .max_glitch_ns = std::min(this->filter_us * 1000u, max_glitch_ns),
     };

